uberfire-extensions/uberfire-commons-editor/uberfire-commons-editor-client/src/test/java/org/uberfire/ext/editor/commons/client/BaseEditorRenameTest.java

```diff
@@ -0,0 +1,110 @@
+import org.uberfire.ext.editor.commons.client.file.RestoreUtil;
+import org.uberfire.ext.editor.commons.client.history.VersionMenuDropDownButton;
+import org.uberfire.ext.editor.commons.client.history.VersionRecordManager;
+import org.uberfire.ext.editor.commons.client.menu.BasicFileMenuBuilder;
+import org.uberfire.ext.editor.commons.client.menu.MenuItems;
+import org.uberfire.ext.editor.commons.version.VersionService;
+import org.uberfire.mocks.CallerMock;
+import org.uberfire.mocks.EventSourceMock;
+import org.uberfire.mvp.PlaceRequest;
+
+import static org.mockito.Matchers.any;
+import static org.mockito.Mockito.*;
+
+@RunWith(GwtMockitoTestRunner.class)
+public class BaseEditorRenameTest {
+
+    private static VersionMenuDropDownButton versionMenuDropDownButton;
+
+    private CallerMock<VersionService> versionServiceCaller;
+
+    private static BaseEditor baseEditor;
+
+    @Mock
+    private static VersionRecordManager versionRecordManager;
+    @Mock
+    private RestoreUtil restoreUtil;
+    @Mock
+    private PlaceRequest placeRequest;
+    @Mock
+    private ClientResourceType clientResourceType;
+    @Mock
+    private BaseEditorView baseView;
+    @Mock
+    private VersionService versionService;
+    @Mock
+    private BasicFileMenuBuilder menuBuilder;
+    @Mock
+    private EventSourceMock<ChangeTitleWidgetEvent> changeTitleNotification = new EventSourceMock<>();
+    @Mock
+    private ObservablePath observablePath;
+
+    @Before
+    public void setup() {
+        versionMenuDropDownButton = new VersionMenuDropDownButton();
+
+        versionServiceCaller = new CallerMock<>(versionService);
+
+        baseEditor = new BaseEditor(versionRecordManager,
+                                    baseView,
+                                    menuBuilder,
+                                    changeTitleNotification) {
+            @Override
+            protected void loadContent() {
+            }
+        };
+        when(restoreUtil.createObservablePath(any(),
+                                              any())).thenReturn(observablePath);
+        baseEditor.init(observablePath,
+                        placeRequest,
+                        clientResourceType,
+                        true,
+                        true,
+                        MenuItems.SAVE,
+                        MenuItems.DELETE,
+                        MenuItems.RENAME,
+                        MenuItems.COPY,
+                        MenuItems.HISTORY);
+    }
+
+    //a reproducer for AF-497
+    @Test
+    public void testVersionRecordManagerIsInitializedOnRename() {
+        //clear the interaction on versionRecordManager from the baseEditor init method
+        Mockito.reset(versionRecordManager);
+
+        when(versionRecordManager.getCurrentPath()).thenReturn(observablePath);
+        baseEditor.onRename();
+
+        verify(versionRecordManager).init(any(),
+                                          any(),
```

Comment:
Wouldn't it be nice to check the path was ```observablePath``` rather than _any_ path?

Reply:
@jsoltes could you please fix this?